std/fmt/parse_float.zig

```diff
@@ -371,22 +355,28 @@ fn caseInEql(a: []const u8, b: []const u8) bool {
     return true;
 }
 
-pub fn parseFloat(comptime T: type, s: []const u8) T {
+pub fn parseFloat(comptime T: type, s: []const u8) !T {
+    if (s.len == 0) {
+        return error.InvalidCharacter;
+    }
+
+    if (s.len >= 2 and s[1] >= 'A') {
```

Comment:
Is there a reason we compare against `2` instead of `3`? All our cases are `>=3` long

Reply:
And we probably want to measure this optimizations impact and leave a comment here explaining why it is worth it do do this check

Reply:
I've ended up removing this optimization for the moment. While it definitely is better assembly it doesn't really factor in to the runtime performance due to the current parsing/conversion dominating the runtime anyway.

I'll leave this for down the track when this is optimized specifically for performance instead.